### PR TITLE
Window styling and Theming API

### PR DESCRIPTION
**Pull request by pchaurasia14**
null

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/wpf/pull/8575)

### CODE REVIEW

**Review by singhashish-wpf — APPROVED** [review]

**Review by dipeshmsft — APPROVED** [review]

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationAccentColorManager.cs [hunk 1]
```diff
@@ -0,0 +1,317 @@
...
+                0xFF
+            );
+        }
+
+        Application.Current.Resources["SystemAccentColor"] = systemAccent;
+        Application.Current.Resources["SystemAccentColorPrimary"] = primaryAccent;
+        Application.Current.Resources["SystemAccentColorSecondary"] = secondaryAccent;
+        Application.Current.Resources["SystemAccentColorTertiary"] = tertiaryAccent;
+
+        Application.Current.Resources["SystemAccentBrush"] = secondaryAccent.ToBrush();
```
Comment (lindexi) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationAccentColorManager.cs:306]: Can I know why do we not use `systemAccent.ToBrush()` to create the `SystemAccentBrush` ?

```suggestion
        Application.Current.Resources["SystemAccentBrush"] = systemAccent.ToBrush();
```

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationThemeManager.cs [hunk 2]
```diff
@@ -0,0 +1,278 @@
...
+    private static ApplicationTheme _cachedApplicationTheme = ApplicationTheme.Unknown;
+
+    internal const string LibraryNamespace = "win11;";
+
+    internal const string ThemesDictionaryPath = "pack://application:,,,/PresentationFramework.Win11;component/Resources/Theme/";
+
+    /// <summary>
+    /// Event triggered when the application's theme is changed.
+    /// </summary>
+    public static event ThemeChangedEvent Changed;
```
Comment (lindexi) [inline src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationThemeManager.cs:47]: Should we use nullable?

```suggestion
    public static event ThemeChangedEvent? Changed;
```

Comment (pchaurasia14) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationThemeManager.cs]: Yes. Had to remove it in order to fix a build issue that complained about use of nullability in nullable contexts.
I've enabled nullability for this file currently.

File: src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationAccentColorManager.cs (hunk 1 above)
Comment (pchaurasia14) [reply src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Appearance/ApplicationAccentColorManager.cs]: Looks like an oversight. Fixed it now.
Thanks

**Pull request merged**